### DISCUSSION TITLE
Hue Gradient Table Lamp

### DISCUSSION THREAD

**Discussion by jb123567** [Koenkk/zigbee2mqtt#15872] [Devices]
@zegl I’ve seen your amazing work with getting native gradient support into Z2M. I have a Hue Gradient Table Lamp which doesn’t seem to have your created options, is there anyway to force the options to appear? Or is it a case of backend changes being needed?

This is the device, https://www.philips-hue.com/en-gb/p/hue-white-and-colour-ambiance-signe-gradient-table-lamp/8718696176238.

**Comment by zegl** [comment]
Hey @jb123567! I've opened a PR that will add Gradient support to your device as well: https://github.com/Koenkk/zigbee-herdsman-converters/pull/5248

**Reply by jb123567** [reply]
Hi @zegl, thank you, have installed, I get all the options but when I try and set a colour, I get "Unknown access "Color in RGB HEX format (eg #663399)"", am I missing something, or shall I raise an issue?

**Reply by zegl** [reply]
The gradient needs to be set via the MQTT API for now, as the frontend does not support setting gradients yet. I’ve opened a PR, but it has not been accepted yet. 

To set a gradient, publish something like `{“gradient”:[“#449966”,”#883399”]}` to the `$NAME/set` topic for your device.

**Reply by jb123567** [reply]
That makes sense, sorry misunderstood, thanks for all your work!!
